core/src/avm1/globals/date.rs

```diff
@@ -599,7 +599,10 @@ fn constructor<'gc>(
     } else {
         this.set_date_time(
             activation.context.gc_context,
-            Some(activation.context.locale.get_current_date_time()),
+            Some(
+                activation.context.locale.get_current_date_time()
+                    + Duration::milliseconds(*activation.context.time_offset as i64),
+            ),
```

Comment:
This could cause a "time travel", where actionscript like this:
```
var t1 = getTimer();
var t2 = new Date();
```
can observe `t2` being earlier than `t1`. (not directly, as `t1` is a time since movie start, but you could observe that by comparing values between frames)
That's why I wrote:
```
All script-visible time sources must be delayed by this amount of ms.
```

Reply:
> in my mind, Date should always match wall time

If we want to present a consistent view to the script and pretend that it did busy-loop for 33ms, then showing Date offset by 33ms would support that "lie".

Reply:
Sure, but I don't think this would actually negatively affect any content -- specifically the "time travel" would only really be visible if you were checking both Date and getTimer inside the spin loop, so it seems to add extra complexity to what is already a hack.  I also suspect this might allow time to run "backwards" once time_offset resets to 0, but a later call to `new Date` returns an "earlier" time (for example, as a result of a mouse listener which runs independent of the framerate), so I don't think it's consistent regardless.

Also consider if we add a general "turbo speed" button. I think we'd still want Date to return the actual wall time in this case, because it's used for get the date for things like calendars, save file time, etc. (Although I suppose this might depend on the specific content, so maybe this is actually a setting).

It's all a minor nit because time_offset gets cleared each frame, and the amount of content it affects is probably 0, so I don't feel too strongly about it in any case.

Reply:
> specifically the "time travel" would only really be visible if you were checking both Date and getTimer inside the spin loop

It's enough to compare "time" before and after the loop.

But anyway, that's a good reasoning; will remove that in a moment.